handlers/controller.js

```diff
@@ -7,28 +11,69 @@ const { models, connectDb } = require ('../dal/database');
  * @returns {Promise<void>} n/a
  */
 let ping = async (req, res) => {
-    res.status(200).send('Ping!');
+    res.status(StatusCode.EASTER_EGG).send('Ping!');
 };
 
 /**
- * Gets product from DAL.
+ * Gets product from DAL - if not found it is retrieved from Barcodelookup and then stored in MongoDB.
  * @param req request.
  * @param res response.
  * @returns {Promise<void>} n/a
  */
 let getProduct = async (req, res) => {
-    let barcodeId = req.params.id;
+    let barcode = req.params.id;
 
     connectDb().then(async () => {
-        models.Product.findOne({ barcodeId })
-            .then((doc) => {
-                if(!doc)
-                    return res.status(404).send();
-                else
-                    res.status(200).send({ doc });
-            }).catch((err) => res.status(400).send(err));
+        try {
+            // search for product in MongoDB
+            let doc = await models.Product.findOne({barcode});
```

Comment:
This is good. In the future we want to send a longer document with a sorted list of manufacturers and the current manufacturer flagged etc. But this is a good step for now.

Reply:
Yep.